.github/workflows/dev.yml

```diff
@@ -124,6 +124,7 @@ jobs:
       - name: Run Release Test
         env:
           ARROW_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
According to the documentation [here](https://docs.github.com/en/actions/using-workflows/using-github-cli-in-workflows), you have to set an environment variable called `GH_TOKEN` in order to use GitHub CLI in a workflow. 

However, setting `GH_TOKEN` may not longer be necessary in this step. At some point `ci/scripts/release_test.sh` was indirectly calling `gh release download` via `dev/release/02-source.sh`. However, I have since changed `dev/release/02-source.sh` to only call `gh release download` if the environment variable `SOURCE_DOWNLOAD` is greater than `0` (I believe @kou suggested this change). 

It looks like `SOURCE_DOWNLOAD` is set `0` in `dev/release/02-source-test.rb` on line [39](https://github.com/apache/arrow/blob/068b9eccd46beb7516283ebc3a784b5866c7bcca/dev/release/02-source-test.rb#L39), which should make setting `GH_TOKEN` unnecessary.  I'll try removing this line from `.github/workflows/dev.yml`.